core/src/main/scala/org/apache/spark/sql/streaming/SnappySinkCallback.scala

```diff
@@ -66,7 +68,12 @@ class SnappyStoreSinkProvider extends StreamSinkProvider with DataSourceRegister
       parameters: Map[String, String],
       partitionColumns: Seq[String],
       outputMode: OutputMode): Sink = {
-    createSinkStateTableIfNotExist(sqlContext)
+    val stateTableSchema = parameters.get(STATE_TABLE_SCHEMA)
+    if (isSecurityEnabled(sqlContext.sparkSession) && stateTableSchema.isEmpty) {
```

Comment:
Can refactor  the if else here and in the called function to match/case

Reply:
Discussed offline. It won't be possible to push this if condition logic inside the `isSecurityEnabled` function and use match case construct there. However, added an optimisation to check the `stateTableSchema` availability prior to checking security in order to avoid unwanted JDBC connection.